src/CameraComponent_V4L2.cpp

```diff
@@ -0,0 +1,467 @@
+        break;
+    case CameraParameters::PARAM_ID_GAIN:
+        ctrl_id = V4L2_CID_GAIN;
+        break;
+    case CameraParameters::PARAM_ID_POWER_LINE_FREQ_MODE:
+        ctrl_id = V4L2_CID_POWER_LINE_FREQUENCY;
+        break;
+    case CameraParameters::PARAM_ID_WHITE_BALANCE_TEMPERATURE:
+        ctrl_id = V4L2_CID_WHITE_BALANCE_TEMPERATURE;
+        break;
+    case CameraParameters::PARAM_ID_SHARPNESS:
+        ctrl_id = V4L2_CID_SHARPNESS;
+        break;
+    case CameraParameters::PARAM_ID_BACKLIGHT_COMPENSATION:
+        ctrl_id = V4L2_CID_BACKLIGHT_COMPENSATION;
+        break;
+    case CameraParameters::PARAM_ID_EXPOSURE_MODE:
+        ctrl_id = V4L2_CID_EXPOSURE_AUTO;
+        break;
+    case CameraParameters::PARAM_ID_EXPOSURE_ABSOLUTE:
+        ctrl_id = V4L2_CID_EXPOSURE_ABSOLUTE;
+        break;
+    case CameraParameters::PARAM_ID_IMAGE_SIZE:
+        ret = setImazeSize(param_value);
+        break;
+    case CameraParameters::PARAM_ID_IMAGE_FORMAT:
+        ret = setImageFormat(param_value);
+        break;
+    case CameraParameters::PARAM_ID_PIXEL_FORMAT:
+        ret = setPixelFormat(param_value);
+        break;
+    case CameraParameters::PARAM_ID_SCENE_MODE:
+        ret = setSceneMode(param_value);
+        break;
+    case CameraParameters::PARAM_ID_VIDEO_SIZE:
+        ret = setVideoSize(param_value);
+        break;
+    case CameraParameters::PARAM_ID_VIDEO_FRAME_FORMAT:
+        ret = setVideoFrameFormat(param_value);
+        break;
+    default:
+        ret = -1;
+        break;
+    }
+
+    if (ctrl_id != -1) {
+        int fd = v4l2_open_device(dev_path.c_str());
+        ret = v4l2_set_control(fd, ctrl_id, param_value);
+        if (ret)
+            log_error("Error in setting control : %s Error:%d", param_id, errno);
+        v4l2_close_device(fd);
+        if (!ret)
+            saveParameter(param_id, param_value);
+    }
+    return ret;
+}
+
+int CameraComponent_V4L2::setParam(const char *param_id, uint8_t param_value)
+{
+    log_debug("%s: Param Id:%s Value:%d", __func__, param_id, param_value);
+    return 0;
+}
+
+int CameraComponent_V4L2::setCameraMode(uint32_t param_value)
+{
+    return 0;
+}
+
+int CameraComponent_V4L2::setImazeSize(uint32_t param_value)
+{
+    return 0;
+}
+
+int CameraComponent_V4L2::setImageFormat(uint32_t param_value)
+{
+    return 0;
+}
+
+int CameraComponent_V4L2::setPixelFormat(uint32_t param_value)
+{
+    return 0;
+}
+
+int CameraComponent_V4L2::setSceneMode(uint32_t param_value)
+{
+    return 0;
+}
+
+int CameraComponent_V4L2::setVideoSize(uint32_t param_value)
+{
+    return 0;
+}
+
+int CameraComponent_V4L2::setVideoFrameFormat(uint32_t param_value)
+{
+    return 0;
+}
+
+int CameraComponent_V4L2::saveParameter(const char *param_id, float param_value)
+{
+    char str[128];
+    mavlink_param_union_t u;
+    u.param_float = param_value;
+    memcpy(&str[0], &u.param_float, sizeof(float));
+    camParam.setParameter(param_id, str);
```

Comment:
Unless param_value is a string disguised as a float, this is not going to work.
memcpy is not going to convert from a float to a char[] and then you call setParameter that is expecting a std::string. If you are going to store the parameter as a float it is better to use a float vector and not a std::string vector. If you are going to use it as a string, it doesn't make sense to cast it to float.
And anyway, if you are going to use strings, you need to ensure '\0' is appended at the end of the string.